nexus/executables/ntest

```diff
@@ -1680,13 +1830,82 @@ def exit_pass(msg=None):
     exit(0)
 #end def exit_pass
 
+# execute function
+from subprocess import Popen,PIPE
+def execute(command,verbose=False,skip=False):
+    out,err = '',''
+    if skip:
+        if verbose:
+            print 'Would have executed:\n  '+command
+        #end if
+    else:
+        if verbose:
+            print 'Executing:\n  '+command
+        #end if
+        out,err = Popen(command,shell=True,stdout=PIPE,stderr=PIPE,close_fds=True).communicate()
+    #end if
+    return out,err
+#end def execute
+
+
+
+def regenerate_reference(update=False,overwrite=True):
+    # create directory for reference file generation
+    refgen_dir = './reference_regen'
+    if os.path.exists(refgen_dir):
+        shutil.rmtree(refgen_dir)
+    #end if
+    os.makedirs(refgen_dir)
+
+    print 'generating reference data in {0}'.format(refgen_dir)
+
+    # create directory for example reference generation
+    refgen_example_dir = os.path.join(refgen_dir,'user_examples')
+    if not os.path.exists(refgen_example_dir):
+        os.makedirs(refgen_example_dir)
+    #end if
+
+    # copy over nexus user examples into reference generation directory
+    execute('rsync -av {0}/* {1}/'.format(example_dir,refgen_example_dir))
```

Comment:
Checking for success would be good here.  A failure to copy the files here will lead to cascading failures that are harder to track down.